ingest/rules/nextclade.smk

```diff
@@ -123,40 +129,109 @@ rule aggregate_gene_coverage_by_gene:
+    benchmark:
+        "benchmarks/v-gen-lab/{gene}/aggregate_gene_coverage_by_gene.txt",
     shell:
         """
         tsv-append -H {input.gene_coverage} > {output.gene_coverage_all}
         """
 
-rule append_gene_coverage_columns:
+rule combine_gene_coverage_columns:
     """
     Append the gene coverage results to the metadata
+    Since gene coverage values should be a value between 0 and 1, empty fields should be filled with 0's
     """
     input:
-        metadata="data/metadata_nextclade.tsv",
+        metadata="data/metadata_all.tsv",
         gene_coverage=expand("results/{gene}/gene_coverage_all.tsv", gene=config["nextclade"]["gene"])
     output:
-        metadata_all="results/metadata_all.tsv",
+        gene_coverage_combined="results/gene_coverage_combined.tsv",
     params:
         id_field=config["curate"]["output_id_field"],
+    log:
+        "logs/v-gen-lab/combine_gene_coverage_columns.txt",
+    benchmark:
+        "benchmarks/v-gen-lab/combine_gene_coverage_columns.txt",
     shell:
         """
-        cp {input.metadata} {output.metadata_all}
+        tsv-select -H -f "{params.id_field}" {input.metadata} > {output.gene_coverage_combined}
         for FILE in {input.gene_coverage}; do
             tsv-join -H \
                 --filter-file $FILE \
                 --key-fields {params.id_field} \
                 --append-fields '*_coverage' \
                 --write-all 0 \
-                {output.metadata_all} \
+                {output.gene_coverage_combined} \
             > results/temp_aggregate_gene_coverage.tsv
-            mv results/temp_aggregate_gene_coverage.tsv {output.metadata_all}
+            mv results/temp_aggregate_gene_coverage.tsv {output.gene_coverage_combined}
         done
         """
 
+rule append_nextclade_columns:
+    """
+    Append the nextclade results to the metadata
+    """
+    input:
+        metadata="data/metadata_all.tsv",
+        genotype_nextclade="results/v-gen-lab/nextclade_metadata.tsv",
+    output:
+        metadata="data/metadata_nextclade.tsv",
+    params:
+        output_nextclade_fields=",".join([f'{value}' for key, value in config["nextclade"]["field_map"].items()][1:]),
+        metadata_id_field=config["curate"]["output_id_field"],
+        nextclade_id_field=config["nextclade"]["id_field"],
+    log:
+        "logs/v-gen-lab/append_nextclade_columns.txt",
+    benchmark:
+        "benchmarks/v-gen-lab/append_nextclade_columns.txt",
+    shell:
+        """
+        augur merge \
+            --metadata \
+                metadata={input.metadata:q} \
+                nextclade={input.genotype_nextclade:q} \
+            --metadata-id-columns \
+                metadata={params.metadata_id_field:q} \
+                nextclade={params.nextclade_id_field:q} \
+            --output-metadata {output.metadata:q} \
+            --no-source-columns \
+        &> {log:q}
+        """
+
+rule append_gene_coverage_columns:
```

Comment:
_non-blocking comment_

`augur merge` can merge more than 2 files at once, so it's potentially possible to combine `append_nextclade_columns` and `append_gene_coverage_columns` into a single rule.

Reply:
Neato! Merged the merge rules as part of https://github.com/nextstrain/dengue/pull/93/commits/c0af2c770c150a3467a1b15ceab15acfe71914c0